Source/WebCore/style/values/transforms/StylePerspective.h

```diff
@@ -31,44 +31,22 @@ namespace Style {
 
 // <'perspective'> = none | <length [0,∞]>
 // https://drafts.csswg.org/css-transforms-2/#propdef-perspective
-struct Perspective {
-    using Length = Style::Length<CSS::Nonnegative, float>;
-
-    Perspective(CSS::Keyword::None) : m_value { } { }
-    Perspective(Length value) : m_value { value } { }
-
-    float usedPerspective() const { return m_value ? std::max(1.0f, m_value->value) : 1.0f; }
-
-    bool isNone() const { return !m_value; }
-    bool isLength() const { return !!m_value; }
-
-    template<typename> bool holdsAlternative() const;
-    template<typename... F> decltype(auto) switchOn(F&&...) const;
-
-    bool operator==(const Perspective&) const = default;
-
-private:
-    template<typename> friend struct Blending;
-
-    Markable<Length> m_value;
+struct Perspective : ValueOrKeyword<Length<CSS::Nonnegative, float>, CSS::Keyword::None> {
+    using Base::Base;
+    using Length = typename Base::Value;
+
+    float usedPerspective() const
+    {
+        if (auto value = tryValue())
+            return std::max(1.0f, value->value);
+        return 1.0f;
+    }
```

Comment:
I’ve seen cases where we use `value_or` for this type of thing. Maybe less efficient.
```suggestion
    float usedPerspective() const { return std::max(1.0f, tryValue().value_or(1.0f)->value); }
```